web-common/src/components/modal/scroll.js

```diff
@@ -0,0 +1,58 @@
+import { getOffset } from "./offset.js";
+const locks = new Set();
+function getScrollbarWidth() {
+  const documentWidth = document.documentElement.clientWidth;
+  return Math.abs(window.innerWidth - documentWidth);
+}
+export function lockBodyScrolling(lockingEl) {
+  locks.add(lockingEl);
+  if (!document.body.classList.contains("sl-scroll-lock")) {
+    const scrollbarWidth = getScrollbarWidth();
+    document.body.classList.add("sl-scroll-lock");
+    document.body.style.setProperty(
+      "--sl-scroll-lock-size",
+      `${scrollbarWidth}px`
+    );
+  }
+}
+export function unlockBodyScrolling(lockingEl) {
+  locks.delete(lockingEl);
+  if (locks.size === 0) {
+    document.body.classList.remove("sl-scroll-lock");
+    document.body.style.removeProperty("--sl-scroll-lock-size");
+  }
+}
+export function scrollIntoView(
```

Comment:
Is there a reason we use this instead of the browsers native scroll into view capability? https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollIntoView

Reply:
Just copied in this file from shoelace, so I don't know if they had a good reason for this approach or not (but in my experience the browser native scroll APIs don't always do exactly what you think they should, and I've had to write wrappers around them in the past, so this doesn't really surprise me).

In either case, digging into it and rewriting+testing it is beyond what I think we should do in this PR, but I could leave a FIXME comment or something if you would like?

Reply:
Nah its cool, I approved despite this